src/main/java/gregtech/common/tileentities/machines/multi/MTEEnvironmentallyControlledChemicalFacility.java

```diff
@@ -0,0 +1,1036 @@
+            drainAmountEU += mPressureEnergyHatch.getBaseMetaTileEntity()
+                .getStoredEU();
+            mPressureEnergyHatch.getBaseMetaTileEntity()
+                .decreaseStoredEnergyUnits(drainAmountEU, false);
+        }
+        if (mMachine && (aTick % 20 == 0)) {
+            if ((currentTemp == 0) && (currentPressure == 0)) {
+                currentPressure = initialPressure;
+                currentTemp = initialTemp;
+            }
+            // Temperature calculation
+            if (isCoolModule || isHeatModule) {
+                // returns values to atmosphere conditions
+                currentTemp = (currentTemp - initialTemp) * COEFF_TEMP + initialTemp;
+                // drain all coolant from hatches and change temperature
+                if (mCoolantInputHatch != null) {
+                    if (mCoolantInputHatch.mFluid != null) {
+                        String coolantName = mCoolantInputHatch.mFluid.getFluid()
+                            .getName();
+                        getCoolantTemperature(coolantName);
+                        boolean isCoolant = ((coolantName.equals("ic2coolant")) || (coolantName.equals("cryotheum"))
+                            || (coolantName.equals("supercoolant"))
+                            || (coolantName.equals("molten.spacetime")));
+                        boolean isHot = ((coolantName.equals("pyrotheum")) || (coolantName.equals("plasma.helium"))
+                            || (coolantName.equals("rawstarmatter"))
+                            || (coolantName.equals("lava")));
+                        // Apply coolant (linear function)
+                        double tempFormulaCoeff = mCoolantInputHatch.mFluid.amount / 10000f * coolantTemp;
+                        if ((isCoolModule || isCoolant) || (isHeatModule || isHot)) {
+                            if (coolantTemp < initialTemp)
+                                currentTemp = Math.max(currentTemp - tempFormulaCoeff, coolantTemp);
+                            if (coolantTemp > initialTemp)
+                                currentTemp = Math.min(currentTemp + tempFormulaCoeff, coolantTemp);
+                            drain(mCoolantInputHatch, mCoolantInputHatch.mFluid, true);
+                        }
+                    }
+                }
+            } else currentTemp = initialTemp;
+
+            // Pressure calculation
+            if (this.isCompressModule || this.isVacuumModule) {
+                // returns values to atmosphere conditions
+                currentPressure = (currentPressure - initialPressure) * COEFF_PRESSURE + initialPressure;
+                // Apply pressure changes
+                if (mPressureEnergyHatch != null) {
+                    drainAmountEU /= 20;
+                    currentPressure += Math.pow(drainAmountEU, 0.7) * (isCompressModule ? 1 : 0);
+                    currentPressure -= Math.pow(drainAmountEU, 0.7) * (isVacuumModule ? 1 : 0);
```

Comment:
```suggestion
                    currentPressure += Math.pow(drainAmountEU, 0.7) * (isCompressModule ? 1 : -1);
```
One of isCompressModule or isVacuumModule _must_ be true at this point due to L995 and the fact that both are mutually exclusive, so can use just a single if statement